app/src/main/java/org/wikipedia/gallery/GalleryActivity.kt

```diff
@@ -0,0 +1,761 @@
+        setNavigationBarColor(Color.BLACK)
+        binding.toolbarGradient.background = GradientUtil.getPowerGradient(R.color.black26, Gravity.TOP)
+        binding.infoGradient.background = GradientUtil.getPowerGradient(R.color.black38, Gravity.BOTTOM)
+        binding.descriptionText.movementMethod = linkMovementMethod
+        binding.creditText.movementMethod = linkMovementMethod
+        (binding.errorView.findViewById<View>(R.id.view_wiki_error_icon) as ImageView)
+            .setColorFilter(ContextCompat.getColor(this, R.color.base70))
+        (binding.errorView.findViewById<View>(R.id.view_wiki_error_text) as TextView)
+            .setTextColor(ContextCompat.getColor(this, R.color.base70))
+        binding.errorView.backClickListener = View.OnClickListener { onBackPressed() }
+        binding.errorView.retryClickListener = View.OnClickListener {
+            binding.errorView.visibility = View.GONE
+            loadGalleryContent()
+        }
+        if (intent.hasExtra(EXTRA_PAGETITLE)) {
+            pageTitle = intent.getParcelableExtra(EXTRA_PAGETITLE)
+        }
+        initialFilename = intent.getStringExtra(EXTRA_FILENAME)
+        revision = intent.getLongExtra(EXTRA_REVISION, 0)
+        sourceWiki = intent.getParcelableExtra(EXTRA_WIKI)!!
+        galleryAdapter = GalleryItemAdapter(this@GalleryActivity)
+        binding.pager.adapter = galleryAdapter
+        binding.pager.registerOnPageChangeCallback(pageChangeListener!!)
+        binding.pager.offscreenPageLimit = 2
+        funnel = GalleryFunnel(app, intent.getParcelableExtra(EXTRA_WIKI), intent.getIntExtra(EXTRA_SOURCE, 0))
+        if (savedInstanceState == null) {
+            if (initialFilename != null) {
+                funnel.logGalleryOpen(pageTitle, initialFilename!!)
+            }
+        } else {
+            controlsShowing = savedInstanceState.getBoolean("controlsShowing")
+            initialImageIndex = savedInstanceState.getInt("pagerIndex")
+            // if we have a savedInstanceState, then the initial index overrides
+            // the initial Title from our intent.
+            initialFilename = null
+            val fm = supportFragmentManager
+            if (supportFragmentManager.backStackEntryCount > 0) {
+                val ft = supportFragmentManager.beginTransaction()
+                for (i in 0 until fm.backStackEntryCount) {
+                    val fragment = fm.findFragmentById(fm.getBackStackEntryAt(i).id)
+                    if (fragment is GalleryItemFragment) {
+                        ft.remove(fragment)
+                    }
+                }
+                ft.commitAllowingStateLoss()
+            }
+        }
+        binding.toolbarContainer.post {
+            if (isDestroyed) {
+                return@post
+            }
+            setControlsShowing(controlsShowing)
+        }
+        if (TRANSITION_INFO != null && TRANSITION_INFO!!.width > 0 && TRANSITION_INFO!!.height > 0) {
```

Comment:
Is it possible to use the `TRANSITION_INFO?.let { ... }` and have the `if else` statement inside to reduce the `!!`?